### PR TITLE
Revise pylint configuration and fix code

### LINKED ISSUE #289: Revise pylint configuration
**Description of issue or feature request**:
We use the [`pylint`](https://www.pylint.org/) configuration file from our [secure-systems-lab/samples-documents](https://github.com/secure-systems-lab/sample-documents/blame/master/pylintrc) repo  (`pylintrc`), which seems to be a generic template from an unknown source that was slightly customized.

However, important checks such as `max-line-length` or `indent-after-paren` don't seem to work.

**Current behavior**:
Over-customized but insufficient `pylintrc`.

**Expected behavior**:
Update/revise `pylintrc` to reliably flag violations of our [code style guide](https://github.com/secure-systems-lab/code-style-guidelines), such as max line length or indentation for continued lines *(and optionally remove unneeded generic checks to make the config more concise)*.

The changes should also be reported back to the [secure-systems-lab/samples-documents](https://github.com/secure-systems-lab/sample-documents/blame/master/pylintrc) repo.

### PR DESCRIPTION
**Pull request by lukpueh**
**Disclaimer:** This PR makes new linter checks a hard test build requirement. I'm happy to discuss, whether this is necessary. IMHO it enhances code quality and reduces repetitive review comments that point to our code style guidelines.

**Fixes issue #**:
Fixes #289 

**Description of the changes being introduced by the pull request**:
Prior to this PR our pylint config file was a full copy from [PyCQA/pylint](https://github.com/PyCQA/pylint/blob/master/pylintrc), listing all defaults explicitly, and adding a few customizations. 

However, the file seems  easier to read and maintain if it only contains the customizations that we require. For instance, prior to this PR, checks like *maximum line length* or *indentation for line continuation* were in one place, but intransparently disabled in another, by adding the check category initials to the `disable` parameter sausage.

The updated config newly enables the following checks:

- whitespace around operators
- line length (<80)
- consider using `in` (instead of e.g. a == "A" or a == "B")
- simplifiable if statement
- `ininstance` instead of `type ==`
- singleton comparison (`is` instead of `==`)
- naming conventions (uppercase constants, snake_case)
- missing docstrings

The code is fixed accordingly. See TODO comments in pylintrc for checks that might be worth adding but require too big of a refactor (e.g. line continuation).

This PR further updates travis/tox to run one separate, parallel lint build (in Python 3.7), and not run the linters in each tested Python version together with the unit tests.

For easy reviewing, please read commit by commit. 

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,9 @@
...
+dist: xenial
 language: python
 
 matrix:
     include:
-        # Explicitly tell Travis which Python version to use for which
-        # tox environment, instead of using one Python version in Travis and
-        # hoping that tox and pyenv run the tests in the desired versions.
-        # https://github.com/travis-ci/travis-ci/issues/8363#issuecomment-355090242
+        # Run each tox env in a travis env that sure has the required Python
```
Comment (adityasaky) [inline .travis.yml]: Should this be:

`Run each tox env in a travis env that has the required Python verison`

File: in_toto/in_toto_keygen.py [hunk 2]
```diff
@@ -113,7 +113,8 @@ def main():
       elif args.type == in_toto.util.KEY_TYPE_ED25519:
         in_toto.util.prompt_generate_and_write_ed25519_keypair(args.name)
       else:  # pragma: no cover
-        log.error("(in-toto-keygen) Unsupported keytype: {0}".format(str(args.type)))
+        LOG.error(
+            "(in-toto-keygen) Unsupported keytype: {0}".format(str(args.type)))
```
Comment (adityasaky) [inline in_toto/in_toto_keygen.py:117]: Side note: I notice we're inconsistent with indexing format parameters. Is it necessary unless we're explicitly messing with the default order? Also, I looked at the code style guidelines and it doesn't speak about `format()` at all, and just about `%s` vs `+` for strings.

**Review by adityasaky — CHANGES_REQUESTED** [review]

File: in_toto/in_toto_keygen.py (hunk 2 above)
Comment (lukpueh) [reply in_toto/in_toto_keygen.py:117]: I think this is okay for now. But feel free to open an issue on the code style repo. ;)

Comment (lukpueh) [reply in_toto/in_toto_keygen.py:117]: For reference, we did have a related discussion in https://github.com/in-toto/in-toto/issues/25.

File: .travis.yml (hunk 1 above)
Comment (lukpueh) [reply .travis.yml]: Fixed in 7fd8c0032342c7049058b5795ba571e5132b2e8a. Let me know if that sounds better.

**Review by adityasaky — APPROVED** [review]
Approved. I noticed a couple of issues were introduced as part of my PRs so apologies for that!

**Pull request merged**